asserts/device_asserts.go

```diff
@@ -44,6 +44,16 @@ func (mod *Model) Model() string {
 	return mod.HeaderString("model")
 }
 
+// ModelDisplayName returns the display-friendly name of the model or
+// falls back to Model if this was not set.
+func (mod *Model) ModelDisplayName() string {
+	display := mod.HeaderString("model-display-name")
```

Comment:
I'd shorten the method to just `DisplayName` anyway (unless there's any chance of other types of display name header on a model). 

As for the header name ... we already have `display-name` for account assertions so maybe we should stick to that for consistency?